pkg/kubelet/stats/pidlimit/pidlimit_linux.go

```diff
@@ -25,19 +25,25 @@ import (
 	"syscall"
 	"time"
 
-	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	statsapi "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
 )
 
 // Stats provides basic information about max and current process count
 func Stats() (*statsapi.RlimitStats, error) {
 	rlimit := &statsapi.RlimitStats{}
 
-	if content, err := ioutil.ReadFile("/proc/sys/kernel/pid_max"); err == nil {
-		if maxPid, err := strconv.ParseInt(string(content[:len(content)-1]), 10, 64); err == nil {
-			rlimit.MaxPID = &maxPid
+	var taskMax int64
+	for _, file := range []string{"/proc/sys/kernel/pid_max", "/proc/sys/kernel/threads-max"} {
+		if content, err := ioutil.ReadFile(file); err == nil {
+			if limit, err := strconv.ParseInt(string(content[:len(content)-1]), 10, 64); err == nil {
+				if taskMax == 0 || taskMax > limit {
+					taskMax = limit
+				}
+			}
```

Comment:
What happens if reads from both files fail?

Reply:
Same as the previous. The MaxPID remains 0 and no error is returned.

Reply:
The previous behavior was keeping the `MaxPID` as `nil`, not initializing it as zero. Can you please clarify?

Reply:
Nice catch. We should leave it as `nil`  because in other places `nil` would be just ignored whereas `0` is used. Corrected.